src/main/java/mate/dao/jdbc/CarJdbcImpl.java

```diff
@@ -0,0 +1,273 @@
+                + "    d.name,"
+                + "    d.license_number "
+                + "FROM"
+                + "    cars AS c"
+                + "        JOIN"
+                + "    manufacturers AS m ON c.manufacturer_id = m.manufacturer_id"
+                + "        JOIN"
+                + "    cars_drivers AS cd ON c.car_id = cd.car_id"
+                + "        JOIN"
+                + "    drivers AS d ON d.driver_id = cd.driver_id "
+                + "WHERE d.driver_id = ? AND c.is_delete = FALSE"
+                + "     AND d.is_delete = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        ResultSet.TYPE_SCROLL_SENSITIVE,
+                        ResultSet.CONCUR_READ_ONLY)) {
+            statement.setObject(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            List<Car> cars = addAllCarsIntoList(resultSet);
+            for (Car car : cars) {
+                car.setDrivers(addAllDriversIntoCar(car.getId(), resultSet));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all by driver by this id " + driverId, e);
+        }
+    }
+
+    private Car createNewCar(ResultSet resultSet) {
+        try {
+            Manufacturer manufacturer = new Manufacturer(resultSet
+                    .getString("manufacturer_name"),
+                    resultSet.getString("manufacturer_country"));
+            manufacturer.setId(resultSet.getObject("manufacturer_id", Long.class));
+            Car car = new Car(resultSet.getString("model"), manufacturer);
+            car.setId(resultSet.getObject("car_id", Long.class));
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car", e);
+        }
+    }
+
+    private List<Car> addAllCarsIntoList(ResultSet resultSet) {
+        try {
+            resultSet.first();
+            List<Car> car = new ArrayList<>();
+            car.add(createNewCar(resultSet));
+            Car flag = car.get(0);
+            resultSet.beforeFirst();
+            while (resultSet.next()) {
+                if (!flag.getId().equals(resultSet.getObject("car_id"))) {
+                    car.add(createNewCar(resultSet));
+                    flag = createNewCar(resultSet);
+                }
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create new car", e);
+        }
+    }
+
+    private List<Driver> addAllDriversIntoCar(Long id, ResultSet resultSet) {
+        try {
+            resultSet.beforeFirst();
+            List<Driver> drivers = new ArrayList<>();
+            while (resultSet.next()) {
+                if (resultSet.getObject("car_id", Long.class).equals(id)) {
```

Comment:
add this to while condition

Reply:
in this case I can't go through whole Result Set, I add just first car and while loop are end